go/host/host.go

```diff
@@ -264,13 +272,16 @@ func (h *host) Unsubscribe(id rpc.ID) {
 }
 
 func (h *host) Stop() {
+	h.logger.Info("Host received a stop command. Attempting shutdown...")
+	h.interrupter.Close()
+	h.shutdownGroup.Wait()
```

Comment:
When you merge in main there's a utility method called `async.WaitTimeout()` that will wait for a WaitGroup with a timeout.

I think if we use that here with a clear error message when it does timeout then it'll help with some of the shutdown hanging issues on e2e tests etc.

When I do the HA work I'm hoping I can figure out what's hanging as a look more closely at the lifecycles of that stuff.